parser/js/src/test/scala/hmda/parser/fi/DateGenerators.scala

```diff
@@ -0,0 +1,15 @@
+package hmda.parser.fi
+
+import org.scalacheck.Gen
+import scalajs.js
+
+object DateGenerators {
+  def randomDate: Gen[Int] = {
+    val beginDate = new js.Date("2017-01-01")
+    val endDate = new js.Date("2020-12-31")
+
+    for {
+      randomDate <- Gen.choose(beginDate.getTime(), endDate.getTime())
+    } yield new js.Date(randomDate).getTime().toInt
```

Comment:
The bug you're encountering with the javascript test is caused here.  The `getTime` method returns a number of milliseconds, not a formatted date.  In turn, this causes the created DAT to be offset on every field after the Date, since this returns a longer number than was expected (with a length of 10 instead of 8).  Which in turn leads to the weird `NumberFormatException`s.